core/virtualhostkeys/virtualhostkeys.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package virtualhostkeys provides unique identifiers
+// for unit and machine host keys. Used for setting
+// and getting virtual host keys from state.
+package virtualhostkeys
+
+type UnitLookup struct {
+	ID string
+}
+
+type MachineLookup struct {
+	ID string
+}
+
+// UnitHostKeyID provides the virtual host key
+// lookup value for a unit based on the unit name.
+func UnitHostKeyID(unitName string) UnitLookup {
```

Comment:
These are very mongo specific and are really only relevant in creating mongo doc ids. As with other such helpers, they belong in state (and are unexported).

Reply:
A question, should I get rid of these all together? I only added them to avoid potential collisions between unit names and machine ID when creating document IDs. But for one, that is probably not possible and/or I can pass in a `names.Tag` and use that string to create the ID.